driver/amdair_chardev.c

```diff
@@ -488,9 +490,21 @@ static int amdair_ioctl_create_queue(struct file *filp, void *data,
 		if (!air_dev || air_dev->device_id != args->device_id)
 			return -ENODEV;
 
-		queue_id = amdair_queue_find_free(air_dev);
+		queue_id = amdair_queue_find_free(air_dev);	
 		if (queue_id == QUEUE_INVALID_ID)
 			return -ENOSPC;
+		
+		amdair_process_get_process_device(air_process, args->device_id,
+						&air_pd);
+
+		// Finding first available slot in datastructure to track
+		// queue IDs used by this process device
+		for(q = 0; q < MAX_HW_QUEUES; q++) {
```

Comment:
Could you move this to a function on the air_process_device? Something like:

amdair_process_device_assign_queue(&air_pd, queue_id);

Also, make sure to add space between for ( and if ( when you move the loop :-).